### PR TITLE
fix(appsync-modelgen-plugin): remove UUID check in java

### LINKED ISSUE #240: Remove UUID check for primary key of Android codegen models and avoid IllegalArgumentException
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-codegen/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-codegen/blob/master/CONTRIBUTING.md#bugs).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v12.18.3

### Amplify CLI Version

5.6.0

### What operating system are you using?

Mac

### Amplify Codegen Command

codegen models

### Describe the bug

A customer reports an issue regarding Android codegen models still requires id to be a UUID. See [https://github.com/aws-amplify/amplify-android/issues/1500](https://github.com/aws-amplify/amplify-android/issues/1500), https://github.com/aws-amplify/amplify-android/issues/846
I used to submitted another [GitHub issue](https://github.com/aws-amplify/amplify-codegen/issues/189) regarding this problem, and a [fix](https://github.com/aws-amplify/amplify-codegen/pull/198) is released. But the fix only removed the try-catch block for checking UUID in `BuildStep` class. The interface of `BuildStep` still throws an `IllegalArgumentException`. And `JustId` method still requires a primary key to be UUID.

### Expected behavior

1. Remove `throws IllegalArgumentException` in BuildStep interface.

**Current Behavior:**
```
public interface BuildStep {
    User build();
    BuildStep id(String id) throws IllegalArgumentException;
}
```

**Expected Behavior:**
```
public interface BuildStep {
    User build();
    BuildStep id(String id);
}
```

2. Remove UUID check in `justId()` method

**Current Behavior:**
```
  /** 
   * WARNING: This method should not be used to build an instance of this object for a CREATE mutation.
   * This is a convenience method to return an instance of the object with only its ID populated
   * to be used in the context of a parameter in a delete mutation or referencing a foreign key
   * in a relationship.
   * @param id the id of the existing item this instance will represent
   * @return an instance of this model with only ID populated
   * @throws IllegalArgumentException Checks that ID is in the proper format
   */
  public static User justId(String id) {
    try {
      UUID.fromString(id); // Check that ID is in the UUID format - if not an exception is thrown
    } catch (Exception exception) {
      throw new IllegalArgumentException(
              "Model IDs must be unique in the format of UUID. This method is for creating instances " +
              "of an existing object with only its ID field for sending as a mutation parameter. When " +
              "creating a new object, use the standard builder method and leave the ID field blank."
      );
    }
    return new User(
      id,
      null,
      null
    );
  }
```

**Expected Behavior:**
```
  /** 
   * WARNING: This method should not be used to build an instance of this object for a CREATE mutation.
   * This is a convenience method to return an instance of the object with only its ID populated
   * to be used in the context of a parameter in a delete mutation or referencing a foreign key
   * in a relationship.
   * @param id the id of the existing item this instance will represent
   * @return an instance of this model with only ID populated
   */
  public static User justId(String id) {
   return new User(
      id,
      null,
      null
    );
  }
```

### Reproduction steps

Run `amplify codegen models` for a graphql schema, where the primary key is a string.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line
type User
    @model
    @auth(rules: [{ allow: owner, operations: [create, delete] }]) {
    id: ID!
    username: String!
    profilePic: String!
    posts: [Post] @connection(fields: ["id"])
}

```

</details>

### Log output

<details>

```
# Put your logs below this line

```

</details>

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by AaronZyLee**
#### Description of changes

- Remove UUID check in justID
- Remove `throws` statement in builder

#### Issue #, if available

fix #240 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] Breaking changes to existing customers are released behind a feature flag or major version update
- [x] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts [hunk 1]
```diff
@@ -12,13 +12,7 @@ import {
...
 import { JAVA_TYPE_IMPORT_MAP } from '../scalars';
 import { JavaDeclarationBlock } from '../languages/java-declaration-block';
-import {
-  AppSyncModelVisitor,
-  CodeGenField,
-  CodeGenModel,
-  ParsedAppSyncModelConfig,
-  RawAppSyncModelConfig,
-} from './appsync-visitor';
+import { AppSyncModelVisitor, CodeGenField, CodeGenModel, ParsedAppSyncModelConfig, RawAppSyncModelConfig } from './appsync-visitor';
```
Comment (AaronZyLee) [inline packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts:15]: lint change

File: packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts [hunk 2]
```diff
@@ -350,12 +344,8 @@ export class AppSyncModelJavaVisitor<
...
    */
   protected generateStepBuilderInterfaces(model: CodeGenModel, isModel: boolean = true): JavaDeclarationBlock[] {
-    const nonNullableFields = this.getWritableFields(model).filter(
-      field => this.isRequiredField(field)
-    );
-    const nullableFields = this.getWritableFields(model).filter(
-      field => !this.isRequiredField(field)
-    );
+    const nonNullableFields = this.getWritableFields(model).filter(field => this.isRequiredField(field));
+    const nullableFields = this.getWritableFields(model).filter(field => !this.isRequiredField(field));
```
Comment (AaronZyLee) [inline packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts:348]: lint change

File: packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts [hunk 3]
```diff
@@ -402,12 +392,8 @@ export class AppSyncModelJavaVisitor<
...
    */
   protected generateBuilderClass(model: CodeGenModel, classDeclaration: JavaDeclarationBlock, isModel: boolean = true): void {
-    const nonNullableFields = this.getWritableFields(model).filter(
-      field => this.isRequiredField(field)
-    );
-    const nullableFields = this.getWritableFields(model).filter(
-      field => !this.isRequiredField(field)
-    );
+    const nonNullableFields = this.getWritableFields(model).filter(field => this.isRequiredField(field));
+    const nullableFields = this.getWritableFields(model).filter(field => !this.isRequiredField(field));
```
Comment (AaronZyLee) [inline packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts:396]: lint change

File: packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts [hunk 4]
```diff
@@ -818,8 +804,8 @@ export class AppSyncModelJavaVisitor<
           default:
             break;
         }
-      })
-    })
+      });
+    });
```
Comment (AaronZyLee) [inline packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts:808]: lint change

**Review by changxu0306 — APPROVED** [review]
lgtm!

**Pull request merged**